web_poet/utils.py

```diff
@@ -160,14 +160,14 @@ def cached_method(method):
     return meth
 
 
-def _cached_method_sync(method, cached_method_name, maxsize=None):
+def _cached_method_sync(method, cached_method_name):
     @wraps(method)
     def inner(self, *args, **kwargs):
         if not hasattr(self, cached_method_name):
             # on a first call, create a lru_cache-wrapped method,
             # and store it on the instance
             bound_method = MethodType(method, self)
-            cached_meth = lru_cache(maxsize=maxsize)(bound_method)
+            cached_meth = lru_cache(maxsize=None)(bound_method)
```

Comment:
I just found out that there is `functools.cache`, equivalent to `functools.lru_cache(maxsize=None)`.

Reply:
Right, but it's 3.9+ only :) Implementation is literally `functools.lru_cache(maxsize=None)` anyways.

Reply:
:facepalm: